server/anonymous/anonymous.go

```diff
@@ -30,17 +35,30 @@ type Config struct {
 // PluginAPI API form mattermost plugin
 type PluginAPI interface {
 	SendEphemeralPost(userID string, post *model.Post) *model.Post
+	GetActivePlugins() ([]PluginIdentifier, error)
 }
 
 type anonymous struct {
 	Config
+	verifiedPlugins map[PluginIdentifier]bool
+
+	unverifiedPluginsList []PluginIdentifier
+	unverifiedPluginsLock *sync.RWMutex // guards
 }
 
 // New returns new Anonymous API object
 func New(apiConfig Config) Anonymous {
-	return &anonymous{
-		Config: apiConfig,
+	return newAnonymous(apiConfig)
+}
+
+func newAnonymous(apiConfig Config) *anonymous {
+	a := &anonymous{
+		Config:                apiConfig,
+		unverifiedPluginsList: []PluginIdentifier{},
+		unverifiedPluginsLock: &sync.RWMutex{},
 	}
+	a.initializeValidatedPackages()
```

Comment:
Again, since these variables are a part of the same logic structure it would be better if they were initialized together, as opposed to one of them being initialized in a separate function.